### PR TITLE
Booking account

### PR DESCRIPTION
**Pull request by zsinx6**
#### What
Add booking accounting.
Make sure to handle the response correctly and with logs.
Small fixes in .gitignore and Makefile.
Removed dead code.

#### Why
We need to account for the current number of booking before creating a new one.

### CODE REVIEW

File: agent/Makefile
Comment (zsinx6) [inline agent/Makefile:39]: For some reason I had problems without this, and it is already there for the backend part. But I'm not sure why it interferes with poetry

Comment (heitorPB) [reply agent/Makefile:39]: Poetry is not POSIX compliant, so you have to force to use Bash instead.

File: agent/lm_agent/workload_managers/slurm/slurmctld_prolog.py
Comment (dusktreader) [inline agent/lm_agent/workload_managers/slurm/slurmctld_prolog.py]: Should this be `logger.error(...)`?

File: agent/lm_agent/workload_managers/slurm/slurmctld_prolog.py
Comment (dusktreader) [inline agent/lm_agent/workload_managers/slurm/slurmctld_prolog.py]: I think you could restructure this without an else as:
```python
if not booking_request:
    logger.error(...)
    sys.exit(1)
logger.debug(success_message)
```

I'm not sure that the explicit sys.exit(0) is necessary. Doesn't python use a 0 by default when a process completes normally?

File: backend/lm_backend/api/booking.py
Comment (dusktreader) [inline backend/lm_backend/api/booking.py]: The language here is a little bit confusing. Maybe "check if the total available is greater than the sum of in_use_total and new_booked."

File: backend/lm_backend/api/booking.py
Comment (dusktreader) [inline backend/lm_backend/api/booking.py]: I don't think the parens are necessary.

**Review by dusktreader — CHANGES_REQUESTED** [review]

File: agent/lm_agent/workload_managers/slurm/slurmctld_prolog.py
Comment (zsinx6) [reply agent/lm_agent/workload_managers/slurm/slurmctld_prolog.py]: I think it is good to be as explicit as possible there

File: backend/lm_backend/api/booking.py
Comment (zsinx6) [reply backend/lm_backend/api/booking.py]: Can you check again now?

File: backend/lm_backend/api/booking.py
Comment (jamesbeedy) [inline backend/lm_backend/api/booking.py]: The use of product_feature and feature don’t match what the variable actually represents, this makes reading and understanding the code less clear. Can we clean up the terminology so that, for example,  a product_feature is a product_feature and a feature is a feature?

File: backend/lm_backend/api/booking.py
Comment (jamesbeedy) [inline backend/lm_backend/api/booking.py]: This too

**Review by dusktreader — APPROVED** [review]
Looks good!

**Comment by jamesbeedy** [comment]
👍 from me. Possibly we wait until after the demo tomorrow to merge this, incase there are any additional adds we may need to make.

**Pull request merged**
